src/shared/bridge.js

```diff
@@ -53,7 +118,7 @@ export default class Bridge {
     const connect = (_token, cb) => {
       if (_token === token) {
         cb();
-        ready();
+        ready(); // This is necessary for testing only.
```

Comment:
I pushed a commit that reworks the tests to avoid the need for the `ready()` callback here. This means that the MessagePort and Window code paths are now the same except that the Window code path checks the token before invoking the callback.

Reply:
I left the code as it was originally (since it will be removed anyway), but I added the comment, because like you said, it didn't look right.